1.(a)
Point(double,double)
Point()
Point(double,double)
x1 = 10.0, y1 = 10.0
x2 = 0.0, y2 = 0.0

(b)
Point.java:5: error: call to this must be first statement in constructor
		this(10,10);
		    ^
1 error

this(10,10); 은 constructor의 맨 앞에 나와야 합니다. 규칙입니다.
this(10,10) must be the first statement of the constructor.

2.(a)
	public double distance(Point other) {
		if (dim() != other.dim()) {
			System.out.println("Error!");
			return 0;
		}
		double sum = 0;
		for(int i = 0; i < vec.length; i++) {
			sum += Math.pow(vec[i] - other.vec[i], 2);
		}
		return Math.sqrt(sum);
	}

(b)
Point(double,double)
Point()
Point(double,double)
distance = 50.0

3.(a)
x = This is MyObj
y = This is SimpleObj

(b)
SomeObj.java:2: error: method does not override or implement a method from a supertype
	@Override
	^
1 error

Override 하는 슈퍼타입이 없음. 
There is an @Override which implies a supertype that should be overrided or have an method to be implemented from the super type.
From the ‘Object’ class, it is trying to use the ‘convertString’ method but this method does not exist so there isn’t a supertype this class has to use this function/method 

4.(a)
Vector@7852e922
inner product = 0.0

(b)
TestFoo.java:6: error: unreported exception Exception; must be caught or declared to be thrown
		double d = v1.product(v2);
		                     ^
1 error

예외가 발생하는 가능성이 있기때문에 throws Exception 아니면 catch 이 있어야 한다. 하지만 여기서 이 예외를 쓰는게 메인 메쏘드에 없다
There is an exception that must be declared to be thrown or caught with ‘throws Exception’ or ‘catch’. However there isn’t any way to show this exception in the main method.

(c)
Vector@7852e922
Exception in thread "main" DimException
	at Vector.product(Vector.java:23)
	at TestVector2.main(TestVector2.java:6)

The dimensions are different with the vectors v1 and v2. So there is an exception shown as the product method will throw DimException if the dimensions of the two vectors are different, which is the requirement for this exception to appear.